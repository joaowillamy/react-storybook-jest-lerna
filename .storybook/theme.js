import { create } from "@storybook/theming/create";

export default create({
  base: "light",

  colorPrimary: "#FDCB13",
  colorSecondary: "#18ACC4",

  // UI
  appBg: "white",
  appContentBg: "#F2F3F4",
  appBorderColor: "#F2F3F4",
  appBorderRadius: 4,

  // Typography
  fontBase: '"Montserrat", sans-serif',
  fontCode: "monospace",

  brandTitle: "Quero front test"
  // brandUrl: 'http://resolvvi.com.br/',
  // brandImage: ''
});
