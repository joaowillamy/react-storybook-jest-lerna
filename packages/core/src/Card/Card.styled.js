import styled, { css } from "styled-components";
import media from "styled-media-query";

export const Card = styled.div`
  padding: 10px;
  box-shadow: ${({ theme }) => theme.shadows.md()};
  transition: all 0.3s;
  display: inline-block;
  width: 90%;
  min-height: 100px;
  text-align: center;
  background-color: #fff;

  img {
    max-width: 100%;
    max-height: 40px;
    margin: 0 auto;
  }

  ${({ hover }) =>
    hover &&
    css`
      cursor: pointer;

      :hover {
        transform: translateY(-2px);
        box-shadow: ${({ theme }) => theme.shadows.lg()};
      }
    `}

  ${media.greaterThan("medium")`
    min-height: 200px;
    width: 23%;
    margin: 0 5px
  `}
`;
