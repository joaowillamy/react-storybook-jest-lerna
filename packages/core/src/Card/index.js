import React from "react";
import PropTypes from "prop-types";

import * as S from "./Card.styled";

const Card = props => {
  return <S.Card {...props} />;
};

Card.propTypes = {
  hover: PropTypes.bool
};

Card.defaultProps = {
  hover: true
};

export default Card;
