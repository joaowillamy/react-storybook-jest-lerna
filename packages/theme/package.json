{
  "name": "@will-ui-kit/theme",
  "version": "1.0.2",
  "main": "lib/index.js",
  "module": "src/index.js"
}
