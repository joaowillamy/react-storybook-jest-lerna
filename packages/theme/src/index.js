export { default as ThemeProvider } from "./ThemeProvider";
export { default as DEFAULT_THEME } from "./theme";
export { GlobalStyles } from "./GlobalStyles";
