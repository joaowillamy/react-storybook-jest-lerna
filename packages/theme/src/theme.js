const BASE = 3;

const borders = {
  sm: color => `1px solid ${color || "#00000029"}`,
  md: color => `3px solid ${color || "#00000029"}`
};

const colors = {
  bluePrimary: "#18ACC4",
  blueSecondary: "#007A8D",
  yellowPrimary: "#FDCB13",
  yellowSecondary: "#DE9E1F",
  green: "#0FA866",
  black: "#1F2D30",
  grey: "#FBFBFB",
  Overlay: "#1F2D30e0" // com 88% de opacidade.
};

const fonts = {
  Montserrat: "Montserrat"
};

const fontSizes = {
  xs: "12px",
  sm: "14px",
  md: "16px",
  lg: "18px",
  xl: "22px",
  xxl: "26px",
  xxxl: "36px"
};

const fontWeights = {
  thin: 100,
  extralight: 200,
  light: 300,
  regular: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  extrabold: 800,
  black: 900
};

const iconSizes = {
  sm: "12px",
  md: "16px",
  lg: "26px"
};

const shadows = {
  sm: color => `0 1px 3px ${color || "#00000029"}`,
  md: color => `0 3px 4px ${color || "#00000029"}`,
  lg: color => `0 3px 6px ${color || "#00000029"}`
};

const space = {
  xs: `${BASE}px`,
  sm: `${BASE * 2}px`,
  md: `${BASE * 3}px`,
  lg: `${BASE * 4}px`,
  xl: `${BASE * 8}px`,
  xxl: `${BASE * 10}px`,
  xxxl: `${BASE * 12}px`
};

const DEFAULT_THEME = {
  borders,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  iconSizes,
  shadows,
  space
};

export default DEFAULT_THEME;
