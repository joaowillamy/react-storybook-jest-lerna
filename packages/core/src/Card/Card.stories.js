import React from "react";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Card from ".";

export default {
  component: Card,
  title: "Core|Card"
};

export const card = () => (
  <ThemeProvider>
    <Card>
      <img
        src={"https://via.placeholder.com/885x307"}
        alt={"Logo quero bolsa"}
      />
      oi
    </Card>
  </ThemeProvider>
);

export const cardWithouHover = () => (
  <ThemeProvider>
    <Card hover={false}>
      <img
        src={"https://via.placeholder.com/885x307"}
        alt={"Logo quero bolsa"}
      />
      oi
    </Card>
  </ThemeProvider>
);
