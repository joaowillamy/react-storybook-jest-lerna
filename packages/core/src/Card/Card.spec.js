import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Card from ".";

describe("Card", () => {
  test("primary", () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <Card>{"Test"}</Card>
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
