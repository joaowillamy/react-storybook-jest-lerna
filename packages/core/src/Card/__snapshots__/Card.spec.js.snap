// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card primary 1`] = `
.c0 {
  padding: 10px;
  box-shadow: 0 3px 4px #00000029;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  display: inline-block;
  width: 90%;
  min-height: 100px;
  text-align: center;
  background-color: #fff;
  cursor: pointer;
}

.c0 img {
  max-width: 100%;
  max-height: 40px;
  margin: 0 auto;
}

.c0:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  box-shadow: 0 3px 6px #00000029;
}

@media (min-width:768px) {
  .c0 {
    min-height: 200px;
    width: 23%;
    margin: 0 5px;
  }
}

<div
  className="c0"
>
  Test
</div>
`;
